xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -618,6 +618,18 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
   }
   // adjust format to the configuration we got
   format.m_channelLayout = GetChannelLayout(format, outconfig.channels);
+  // we might end up with an unusable channel layout that only contains
+  // UNKNOWN1 channels, let's do a sanity check
+  bool isSaneLayout = false;
+  for (unsigned int i = 0; i < format.m_channelLayout.Count(); ++i)
+  {
+    isSaneLayout = (format.m_channelLayout[i] != AE_CH_UNKNOWN1);
```

Comment:
ALSA codepath is not what counts here. Important is why we have something like UNKNOWN and how it can be used. We should always test against the spec and not against the current code that may change.